src/UnigramWordPredictor.java

```diff
@@ -51,7 +53,18 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    Map<String, List<String>> neighborMap = new HashMap<>();
+    for (int i =0; i < trainingWords.size() -1; i++){
+      String word = trainingWords.get(i);
+      String next = trainingWords.get(i + 1);
+      if((neighborMap.isEmpty())||(!neighborMap.containsKey(word))){
+        neighborMap.put(word, (new ArrayList<>(Arrays.asList(next))));
+      } else {
+        neighborMap.get(word).add(next);
+      }
+    }
```

Comment:
Great logic! I think you can simplify line 60 a bit though; you only need the second condition in the `if`. Can you consider why?